server/api/crud/pipelines.py

```diff
@@ -433,3 +440,25 @@ def _get_experiment_id_from_run(run: dict) -> str:
             ):
                 return data.get("id", "")
         return ""
+
+    def _filter_runs_by_name(self, runs: list, target_name: str) -> list:
+        """Filter runs by their name while ignoring the project string on them
+
+        :param runs: list of runs to be filtered
+        :param target_name: string that should be part of a valid run name
+        :return: filtered list of runs
+        """
+        if not target_name:
+            return runs
+
+        def filter_by(run):
+            run_name = run.get("name", "")
+            if self.resolve_project_from_pipeline(run) != "*":
```

Comment:
I don't think `resolve_project_from_pipeline` returns `*` under any scenario. I believe the case you try to handle here is the situation where the function doesn't return anything?

Reply:
Good one, Saar!

This, however, brings us to a gap in the task requirements. Do we want `name-contains` not to consider the project's name even when the API request specifies all projects? If so, my last commit should clear any remaining problems. If not, we might need to pass `project_name` to this function after all.

What do you think would be best?

Reply:
@laurybueno - indeed, the filtering logic should not be affected by the list-query scope. From user perspective, the workflow name is the part that's after the project name prefix (which is how the user called the workflow to begin with), and this shouldn't change because the query was for all projects. So, each workflow name should always be stripped of its respective project name before doing the `name-contains` filter. The only exception is the case where the workflow name was originally empty, in which case we assume the workflow name is the project name (not performing stripping). 
Looks like your new code fits this description (with a small glitch, see my new comment).